server.go

```diff
@@ -3424,6 +3440,80 @@ func (s *server) findPeerByPubStr(pubStr string) (*peer.Brontide, error) {
 	return peer, nil
 }
 
+func (s *server) sendBackupToPeer(data []byte) error {
+	serverPeers := s.Peers()
+
+	if len(data) > lnwire.MaxPeerStorageBytes {
+		data = data[:lnwire.MaxPeerStorageBytes+1]
+	}
+
+	for _, p := range serverPeers {
+		p.SendBackupMtx.Lock()
+		if p.RemoteFeatures().HasFeature(
+			lnwire.OptionProvideStorageOptional) &&
+			p.SendBackup == 0 {
+
+			if err := p.SendMessage(false, &lnwire.PeerStorage{
+				Blob: data,
+			}); err != nil {
+				return fmt.Errorf("unable to send backup "+
+					"storage to peer(%v), received error: "+
+					"%v", p.PubKey(), err)
+			}
+			p.SendBackup = 1
+			p.LatestPeerBackup = data
+
+		}
+		p.SendBackupMtx.Unlock()
+	}
+
+	return nil
+}
```

Comment:
This is absolutely not what we want to do. It opaquely truncates the data passed in (which is not something we should do without warning. It also seems to send things when the SendBackup flag is *disabled*. Finally it replicates the same data across all peers.

Reply:
I would add a log message for the warning, when sendBackup flag is zero it is not disabled, if there is any point of confusion in the code that contradicts, it should be corrected by me when I submit another iteration for this.